training/src/main/java/com/wolox/training/route/BookRoute.java

```diff
@@ -0,0 +1,82 @@
+package com.wolox.training.route;
+
+import com.wolox.training.dto.BookInputDto;
+import com.wolox.training.dto.BookResponseDto;
+import com.wolox.training.exceptions.BookException;
+import com.wolox.training.processor.ErrorProcessor;
+import com.wolox.training.services.BookService;
+import org.apache.camel.Exchange;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.cdi.ContextName;
+import org.apache.camel.model.rest.RestBindingMode;
+import org.apache.camel.model.rest.RestParamType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@ContextName("books")
+public class BookRoute extends RouteBuilder {
+
+  @Autowired
+  private BookService bookService;
+
+  @Override
+  public void configure() throws Exception {
+
+    onException(BookException.class).handled(true)
+        .setHeader(Exchange.CONTENT_TYPE, constant("text/plain")).process(new ErrorProcessor());
+
+    restConfiguration()
+        .bindingMode(RestBindingMode.json)
+        .apiContextPath("/api")
+        .contextPath("/rest")
+        .apiContextPath("/api")
+        .contextPath("/rest")
+        .apiProperty("host", "")
+        .dataFormatProperty("prettyPrint", "true")
+        .enableCORS(true)
+
+        .apiProperty("api.title", "API Book")
+        .apiProperty("api.contact.name", "Jhovanny Canas").apiProperty("api.version", "0.0.1");
+
+    rest().description("Wolox Training Book Service")
+        .consumes("application/json")
+        .produces("application/json")
+
+        .get("/books").description("allows you to consult all the books")
+        .outType(BookResponseDto[].class)
+        .responseMessage().code(200).message("OK").endResponseMessage()
+        .responseMessage().code(404).message("no books found").endResponseMessage()
+        .responseMessage().code(500).message("error generating query").endResponseMessage().route()
+        .streamCaching().bean(this.bookService, "getAllBook").endRest()
```

Comment:
This kind of definition could be more clear if we use in a better way the spaces and line breaks:
```
        .get("/books")
            .description("allows you to consult all the books")
            .outType(BookResponseDto[].class)
            .responseMessage().code(200).message("OK").endResponseMessage()
            .responseMessage().code(404).message("no books found").endResponseMessage()
            .responseMessage().code(500).message("error generating query").endResponseMessage().route()
            .streamCaching().bean(this.bookService, "getAllBook")
            .endRest()
```